### PR TITLE
Add support for C GLib

### LINKED ISSUE #154: Add C glib builders
Steps (these could be documented in `projects/arrow/README.md`):

1. Add a new docker image with the required dependencies added, based on the [cpp one](https://github.com/ursa-labs/ursabot/blob/master/projects/arrow/arrow/docker.py#L41). 
2. Build the image(s) via `ursabot docker -n c_glib build` (assuming that the image is named as `c_glib`).
3. Add a builder based on the [CppTest builder](https://github.com/ursa-labs/ursabot/blob/master/projects/arrow/arrow/builders.py#L302) with c_glib specific steps and properties.
4. Add it to arrow's [master.cfg](https://github.com/ursa-labs/ursabot/blob/master/projects/arrow/master.cfg#L73-L104)
5. [Configure a scheduler](https://github.com/ursa-labs/ursabot/blob/master/projects/arrow/master.cfg#L161-L165) with the appropiate file change paths (`cpp/*` and `c_glib/*`)
6. Add an entry for the builder in the [.travis.yml](https://github.com/ursa-labs/ursabot/blob/master/.travis.yml#L121-L125) to test it automatically.

@kou would you like to pick up this issue?

### PR DESCRIPTION
**Pull request by kou**
- fixes #154
- also enabled Gandiva in the plain ubuntu builders

### CODE REVIEW

**Comment by kszucs** [comment]
@kou this looks great, thanks for doing it! I'll review (and probably fix the flake8 issues) tomorrow.

**Comment by kszucs** [comment]
Travis builds are redundant with the azure pipelines builds. Azure is green, merging.

File: projects/arrow/arrow/builders.py [hunk 1]
```diff
@@ -190,6 +190,35 @@
...
+    j=util.Property('ncpus', 6),
+    name='Compile C GLib',
+    workdir='c_glib/build'
+)
+c_glib_install = Ninja(
+    'install',
+    name='Install C GLib',
+    workdir='c_glib/build'
+)
+c_glib_install_test_dependencies = Bundle(
```
Comment (kszucs) [inline projects/arrow/arrow/builders.py]: We can spare the test dependency installation time if we move `bundle` to the docker image, created issue https://github.com/ursa-labs/ursabot/issues/174

**Comment by kou** [comment]
Thanks for completing this!

**Pull request merged**